lib/core/test/integration/Test/Integration/Scenario/CLI/Wallets.hs

```diff
@@ -142,7 +167,52 @@ spec = do
         err `shouldBe` "Ok.\n"
         out `shouldNotContain` "CLI Wallet"
         c `shouldBe` ExitSuccess
-
   where
     emptyWallet' :: Context t -> IO String
     emptyWallet' = fmap (T.unpack . view walletId) . emptyWallet
+
+specPorts :: Int -> SpecWith (Context t)
+specPorts defaultPort = do
+    it "PORTS_01 - Can create wallet with specifying port" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            (Just (show defaultPort))
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "Ok.\n")
+            (`shouldContain` walletName)
+    it "PORTS_02 - Can create wallet with relying on default port without specifying it" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            Nothing
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "Ok.\n")
+            (`shouldContain` walletName)
+    it "PORTS_03 - Cannot create wallet with wrong port" $ \_ -> do
+        let walletName = "Wallet created via CLI"
+        checkSimpleCreateWallet
+            (Just (show (defaultPort + 1)))
+            "15"
+            walletName
+            (`shouldBe` ExitSuccess)
+            (`shouldContain` "does not exist (Connection refused))\n")
+            (`shouldNotContain` walletName)
+
+checkSimpleCreateWallet
+    :: Maybe String
+    -> String
+    -> String
+    -> (ExitCode -> IO ())
+    -> (String -> IO ())
+    -> (String -> IO ())
+    -> IO ()
+checkSimpleCreateWallet portM mnemonicSize walletName exitCodeCheck errCheck outCheck = do
+    Stdout mnemonics <- generateMnemonicsViaCLI ["--size", mnemonicSize]
+    let passphrase = "Secure passphrase"
+    (c, out, err) <- createWalletViaCLI [walletName] mnemonics "\n" passphrase portM
+    exitCodeCheck c
+    errCheck (T.unpack err)
+    outCheck out
```

Comment:
I understand that this check just landed here in this method, because it was there in the code in the first place :) Actually it is quite poor and I was planning to add more detailed checks (like here: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core/test/integration/Test/Integration/Scenario/CLI/Transactions.hs#L92-L97) but didn't get there yet for `Wallets.sh`. (ofc that's not in scope of this pr)

For that matter, I'd be rather for separating action and assertion not to be in a single method (but perhaps, that's just a personal preference)